src/replay/service.zig

```diff
@@ -31,151 +36,492 @@ const MAX_ENTRIES: u64 = 1024 * 1024; // 1 million slots is about 5 days
 const DUPLICATE_LIVENESS_THRESHOLD: f64 = 0.1;
 pub const DUPLICATE_THRESHOLD: f64 = 1.0 - SWITCH_FORK_THRESHOLD - DUPLICATE_LIVENESS_THRESHOLD;
 
+pub const Logger = sig.trace.ScopedLogger("replay");
+
 pub const ReplayDependencies = struct {
     /// Used for all allocations within the replay stage
     allocator: Allocator,
-    logger: sig.trace.Logger,
-    my_identity: sig.core.Pubkey,
+    logger: Logger,
+    my_identity: Pubkey,
+    vote_identity: Pubkey,
     /// Tell replay when to exit
     exit: *std.atomic.Value(bool),
     /// Used in the EpochManager
     epoch_schedule: sig.core.EpochSchedule,
+    account_store: sig.accounts_db.AccountStore,
+    /// Reader used to get the entries to validate them and execute the transactions
+    /// Writer used to update the ledger with consensus results
+    ledger: LedgerRef,
     /// Used to get the entries to validate them and execute the transactions
-    blockstore_reader: *BlockstoreReader,
-    /// Used to update the ledger with consensus results
-    ledger_result_writer: *LedgerResultWriter,
-    account_store: AccountStore,
     slot_leaders: SlotLeaders,
+    senders: Senders,
+    receivers: Receivers,
+
     /// The slot to start replaying from.
-    root_slot: Slot,
-    root_slot_constants: sig.core.SlotConstants,
-    root_slot_state: sig.core.SlotState,
+    root: struct {
+        slot: Slot,
+        constants: sig.core.SlotConstants,
+        state: sig.core.SlotState,
+    },
 };
 
-pub const Logger = sig.trace.ScopedLogger("replay");
+pub const LedgerRef = struct {
+    db: BlockstoreDB,
+    reader: *sig.ledger.BlockstoreReader,
+    writer: *sig.ledger.LedgerResultWriter,
+};
+
+pub const SlotHistoryAccessor = struct {
```

Comment:
My instinct would be to define this where it's used (replay_tower.zig). What do you think?

Reply:
Actually had a TODO on this regarding what the best place might. `replay_tower.zig` sounds ok.

[d5af195](https://github.com/Syndica/sig/pull/810/commits/d5af1950c9c25c61124df41510c5ea749db4cd84)